zallet/src/components/json_rpc/methods/get_transaction.rs

```diff
@@ -0,0 +1,360 @@
+
+    /// The time in seconds since epoch (1 Jan 1970 GMT).
+    #[serde(skip_serializing_if = "Option::is_none")]
+    blocktime: Option<u64>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    expiryheight: Option<u64>,
+
+    walletconflicts: Vec<String>,
+
+    /// The transaction time in seconds since epoch (1 Jan 1970 GMT).
+    time: u64,
+
+    /// The time received in seconds since epoch (1 Jan 1970 GMT).
+    timereceived: u64,
+
+    details: Vec<Detail>,
+
+    /// Raw data for transaction.
+    hex: String,
+}
+
+#[derive(Clone, Debug, Serialize, JsonSchema)]
+struct Detail {
+    /// The Zcash address involved in the transaction.
+    address: String,
+
+    /// The category.
+    ///
+    /// One of 'send' or 'receive'.
+    category: String,
+
+    /// The amount in ZEC.
+    amount: f64,
+
+    /// The amount in zatoshis.
+    #[serde(rename = "amountZat")]
+    amount_zat: u64,
+
+    /// The vout value.
+    vout: u64,
+}
+
+pub(super) const PARAM_TXID_DESC: &str = "The ID of the transaction to view.";
+pub(super) const PARAM_INCLUDE_WATCHONLY_DESC: &str =
+    "Whether to include watchonly addresses in balance calculation and `details`.";
+pub(super) const PARAM_VERBOSE_DESC: &str = "Must be `false` or omitted.";
+pub(super) const PARAM_AS_OF_HEIGHT_DESC: &str = "Execute the query as if it were run when the blockchain was at the height specified by this argument.";
+
+pub(crate) async fn call(
+    wallet: &DbConnection,
+    chain: FetchServiceSubscriber,
+    txid_str: &str,
+    include_watchonly: bool,
+    verbose: bool,
+    as_of_height: Option<i64>,
+) -> Response {
+    let txid = parse_txid(txid_str)?;
+
+    let filter = if include_watchonly {
+        is_mine_spendable_or_watchonly
+    } else {
+        is_mine_spendable
+    };
+
+    if verbose {
+        return Err(LegacyCode::InvalidParameter.with_static("verbose must be set to false"));
+    }
+
+    let as_of_height = parse_as_of_height(as_of_height)?;
+
+    // Fetch this early so we can detect if the wallet is not ready yet.
+    let chain_height = wallet
+        .chain_height()
+        .map_err(|e| LegacyCode::Database.with_message(e.to_string()))?
+        .ok_or_else(|| LegacyCode::InWarmup.with_static("Wait for the wallet to start up"))?;
+
+    let tx = wallet
+        .get_transaction(txid)
+        .map_err(|e| {
+            LegacyCode::Database
+                .with_message(format!("Failed to fetch transaction: {}", e.to_string()))
+        })?
+        .ok_or(LegacyCode::InvalidParameter.with_static("Invalid or non-wallet transaction id"))?;
+
+    // TODO: In zcashd `filter` is for the entire transparent wallet. Here we have multiple
+    // mnemonics; do we have multiple transparent buckets of funds?
+
+    // `gettransaction` was never altered to take wallet shielded notes into account.
+    // As such, its `amount` and `fee` fields are calculated as if the wallet only has
+    // transparent addresses.
```

Comment:
blocking: This has been a potential source of loss of funds for years, and we should not reproduce these bugs. Instead of modifying the semantics of this method, we should not include it in the `Zallet` alpha, and instead provide a replacement in `z_viewtransaction` that we can encourage users to migrate to instead.